00-programming-fundamentals/intro-to-scripts.md

```diff
@@ -0,0 +1,72 @@
+# Introduction to Scripts
+
+## What is a script?
+
+By now you should be familiar with a variety of different commands from the terminal.  Things like `ls` and `touch`.  However, it turns out macOS doesn't come with _every_ command you could ever want.  Things like the `code` and `brew` commands are once you added yourself.
+
+A "script" is a command written in a "Scripting Language," luckily you already know one, Ruby!  Scripting languages are special in that they are designed to be easy to read _and_ run.  If you were to open the `ls` command all you are would see is a bunch of gibberish.
+
+## How to run a script
+
+Start by downloading this [demo script](./demo-script.rb)!
+
+You should already be familiar with the simplest way to run a Ruby script.  You can `ruby ./demo-script.rb`.  If a script is properly formatted with a magic comment at the top you can run it with `./demo-script-rb` leaving off the `ruby` from the front.
```

Comment:
Don't need the trailing `-rb`

Reply:
That was a brain fart, it was suppose to say `.rb`.  I just wanted to be clear that it was a Ruby file.  Do you think when testing that we should have the demo script just be `demo-script` without the extension?

Reply:
Ah, I see. I would stick with `demo-script.rb`.